testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -103,10 +123,10 @@ public void testTrinoReadingSparkData()
         onSpark().executeQuery("DROP TABLE " + sparkTableName);
     }
 
-    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
-    public void testSparkReadingTrinoData()
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS}, dataProviderClass = TestingIcebergStorageFormat.class, dataProvider = "storageFormats")
+    public void testSparkReadingTrinoData(TestingIcebergStorageFormat storageFormat)
     {
-        String baseTableName = "test_spark_reading_primitive_types";
+        String baseTableName = "test_spark_reading_primitive_types_" + storageFormat.name();
```

Comment:
Maybe add a method
```java
String baseTableName = baseTableName("test_spark_reading_primitive_types", storageFormat);
```
I don't feel strongly but seems a bit nicer, and easier to change later